dependencies.yaml

```diff
@@ -24,5 +24,5 @@ protobuf: 3.20.0
 zlib: 1.2.13
 zstd: 1.5.2
 snappy: 1.1.9
-openssl: 1.1.1q
+openssl: 1.1.1s
```

Comment:
Why did you upgrade the OpenSSL version? It's okay to add a patch to an existing OpenSSL release like the line removed in this PR.

```bash
        echo -e "#include <string.h>\n$(cat test/v3ext.c)" > test/v3ext.c
```

It's better to use a separate PR to upgrade the dependency version.

Reply:
This is an OpenSSL bug, It was fixed in 1.1.1s, I fixed it by the way.

https://github.com/openssl/openssl/issues/18720#issuecomment-1327885780

Reply:
It's just a compilation bug. The previous scripts already fixes it by adding the patch to include the `string.h`.

Reply:
There is also a bug when building Python 3.7 from source, adding a patch is a common used fixed method. https://github.com/apache/pulsar-client-python/blob/c5c177af647997d1182f3d4a69c74824f39af38f/pkg/mac/build-dependencies.sh#L73

Though it's okay to upgrade the OpenSSL dependency, but it's not related to this PR.

Reply:
Okay, I submit in #251.